src/sentry/notifications/utils/__init__.py

```diff
@@ -129,6 +130,30 @@ def summarize_issues(
     return rv
 
 
+def get_email_link_extra_params(
+    referrer: str = "alert_email",
+    rule_details: Sequence[NotificationRuleDetails] | None = None,
+    alert_timestamp: int | None = None,
+) -> dict[int, str]:
+    alert_timestamp_str = (
+        str(round(time.time() * 1000)) if not alert_timestamp else str(alert_timestamp)
+    )
+    return {
+        rule_detail.id: "?"
+        + str(
+            urlencode(
+                {
+                    "referrer": referrer,
+                    "alert_type": str(AlertRuleTriggerAction.Type.EMAIL.name).lower(),
+                    "alert_timestamp": alert_timestamp_str,
+                    "alert_rule_id": rule_detail.id,
+                }
+            )
+        )
+        for rule_detail in rule_details
```

Comment:
Type error when `rule_details` parameter is `None` so either short-circuit at the top or do: 
```py
for rule_detail in (rule_details or [])
```

Reply:
Hah, I like the second option. Will do.